### PR TITLE
bpo-34512: Document platform-specific strftime() behavior for non-ASC…

### PR DESCRIPTION
**Pull request by izbyshev**
…II format strings

https://bugs.python.org/issue34512

### CODE REVIEW

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -2029,6 +2029,12 @@ calls the platform C library's :func:`strftime` function, and platform
 variations are common.  To see the full set of format codes supported on your
 platform, consult the :manpage:`strftime(3)` documentation.
 
+Likewise, handling of format strings containing Unicode code points that can't
+be represented in the charset of the current locale is platform-dependent. On
+some platforms such code points are preserved intact in the output, while on
+others ``strftime`` may raise :exc:`UnicodeError` or return an empty string
+instead.
+
```
Comment (izbyshev) [inline Doc/library/datetime.rst:2037]: I haven't mentioned surrogate code points and `surrogateescape` error handler. Not sure whether we want to go to that level of detail.

`time` module docs should also be changed when/if we reach consensus on the wording.

**Comment by pganssle** [comment]
Presumably this can be backported as far back as is supported at the moment, since this is a long-standing issue.

File: Doc/library/datetime.rst [hunk 2]
```diff
@@ -2029,6 +2029,12 @@ calls the platform C library's :func:`strftime` function, and platform
 variations are common.  To see the full set of format codes supported on your
 platform, consult the :manpage:`strftime(3)` documentation.
 
+Likewise, handling of format strings containing Unicode code points that can't
```
Comment (taleinat) [inline Doc/library/datetime.rst]: Suggestion: "Likewise" -> "For the same reason"

File: Doc/library/datetime.rst [hunk 3]
```diff
@@ -2029,6 +2029,12 @@ calls the platform C library's :func:`strftime` function, and platform
...
+be represented in the charset of the current locale is platform-dependent. On
```
Comment (taleinat) [inline Doc/library/datetime.rst]: "... is also platform-dependent."

File: Doc/library/datetime.rst (hunk 1 above)
Comment (taleinat) [reply Doc/library/datetime.rst:2037]: I tend to agree that we should avoid mentioning surrogates given the proposed wording.

**Comment by izbyshev** [comment]
@taleinat Thank you for the review! I've made the requested changes.

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no longer accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Review by pganssle — APPROVED** [review]

**Review by taleinat — APPROVED** [review]

**Pull request merged**